src/zenml/zen_stores/schemas/pipeline_run_schemas.py

```diff
@@ -216,7 +216,7 @@ def to_model(self, hydrate: bool = False) -> "PipelineRunResponse":
         }
 
         if self.deployment is not None:
-            steps = {s.name: s.to_model() for s in self.deployment.step_runs}
+            steps = {s.name: s.to_model() for s in self.step_runs}
```

Comment:
@bcdurak I wanted to ask anyway, this is not related to any request optimization but just a bug I found I think. Or is there any reason a pipeline run should have all the steps associated with a deployment, which could contain multiple runs?

Reply:
The `if-elif` condition is there to fix some compatibility issues. I do not remember the exact case off the top of my head, but it is possible that our DB has pipeline runs in two different shapes or forms. 

- The first one is the current version of the runs with a `deployment_id`, thus stack, build, etc, can be inferred through the deployment.
- The second one is the outdated pipelines which do not have a deployment_id but stack_id, build_id, etc. So, we use them to infer these values.
- In any case, one of these conditions must apply for this to work. So, it will be computed only once but for the sake of cleaning the code, you can move it out if you want to.

On a separate account, @schustmi, you are right. It is nice that we caught this and fixed it.

Reply:
The general if-elif is fine, but the steps are just plain wrong right? If we run any scheduled pipeline right now, I'd imagine the runs will have the combined steps of all the runs.

Reply:
I just checked, and not sure the actual result is worse or better: Due to the same name and this being a dictionary, all runs of a deployment contain the same steps (the ones of the latest run)

Reply:
Your fix is definitely better.